src/Microsoft.Azure.NotificationHubs/KeyValueConfigurationManager.cs

```diff
@@ -106,10 +122,208 @@ private static NameValueCollection CreateNameValueCollectionFromConnectionString
             }
         }
+
+        public NamespaceManager CreateNamespaceManager()
+        {
+            Validate();
+
+            string operationTimeout = connectionProperties[OperationTimeoutConfigName];
+            IEnumerable<Uri> endpoints = GetEndpointAddresses(connectionProperties[EndpointConfigName], connectionProperties[ManagementPortConfigName]);
+            IEnumerable<Uri> stsEndpoints = GetEndpointAddresses(connectionProperties[StsEndpointConfigName], null);
+            string issuerName = connectionProperties[SharedSecretIssuerConfigName];
+            string issuerKey = connectionProperties[SharedSecretValueConfigName];
+            string sasKeyName = connectionProperties[SharedAccessKeyName];
+            string sasKey = connectionProperties[SharedAccessValueName];
+            string windowsDomain = connectionProperties[WindowsDomainConfigName];
+            string windowsUsername = connectionProperties[WindowsUsernameConfigName];
+            SecureString windowsPassword = GetWindowsPassword();
+            string oauthDomain = connectionProperties[OAuthDomainConfigName];
+            string oauthUsername = connectionProperties[OAuthUsernameConfigName];
+            SecureString oauthPassword = GetOAuthPassword();
+
+            try
+            {
+                TokenProvider provider = CreateTokenProvider(stsEndpoints, issuerName, issuerKey, sasKeyName, sasKey, windowsDomain, windowsUsername, windowsPassword, oauthDomain, oauthUsername, oauthPassword);
+                if (string.IsNullOrEmpty(operationTimeout))
+                {
+                    return new NamespaceManager(endpoints, provider);
+                }
+
+                return new NamespaceManager(
+                    endpoints,
+                    new NamespaceManagerSettings()
+                    {
+                        TokenProvider = provider
+                    });
+            }
+            catch (ArgumentException e)
+            {
+                throw new ConfigurationErrorsException(
+                    SRClient.AppSettingsCreateManagerWithInvalidConnectionString(e.Message),
+                    e);
+            }
+            catch (UriFormatException e)
+            {
+                throw new ConfigurationErrorsException(
+                    SRClient.AppSettingsCreateManagerWithInvalidConnectionString(e.Message),
+                    e);
+            }
+        }
+
+        public SecureString GetWindowsPassword()
+        {
+            return GetSecurePassword(WindowsPasswordConfigName);
+        }
+
+        public SecureString GetOAuthPassword()
+        {
+            return GetSecurePassword(OAuthPasswordConfigName);
+        }
+
+        internal TokenProvider CreateTokenProvider()
+        {
+            IList<Uri> endpointAddresses = GetEndpointAddresses(connectionProperties["StsEndpoint"], (string)null);
+            string connectionProperty1 = connectionProperties["SharedSecretIssuer"];
```

Comment:
Consider `var` versus declaration `string` or `SecureString` here

Reply:
change to var for all variables in this method